etc/cmake/options.cmake

```diff
@@ -41,6 +41,16 @@ if(OT_BORDER_ROUTER)
     list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1")
 endif()
 
+option(OT_NCP "enable network co-processor NCP")
```

Comment:
Did you prepare a PR towards the upstream OpenThread repository? Ideally, we should get this change from upstream, I'd rather avoid local changes to OT code base.

Please split this change into a separate commit, add `[ot fromlist]` prefix to the commit title and add a link to the upstream PR introducting this change in the commit message. We'll need to revert it before next upstream merge.

Reply:
> Did you prepare a PR towards the upstream OpenThread repository? Ideally, we should get this change from upstream, I'd rather avoid local changes to OT code base.

No, I did not. Trying to get initial feedback on the complete package.

> Please split this change into a separate commit, add `[ot fromlist]` prefix to the commit title and add a link to the upstream PR introducting this change in the commit message. We'll need to revert it before next upstream merge.

Will do.

Thanks for immediate review. Please also have a look at https://github.com/zephyrproject-rtos/zephyr/pull/23249.

Reply:
I will, I need to allocate more time for that though :)

Reply:
Created https://github.com/openthread/openthread/pull/4633 in upstream OpenThread and rebased the branch.